src/decoder/mod.rs

```diff
@@ -1505,10 +1546,9 @@ impl<R: Read + Seek> Decoder<R> {
 
     /// Decodes the entire image and return it as a Vector
     pub fn read_image(&mut self) -> TiffResult<DecodingResult> {
-        let result = match (self.image().chunk_type, self.image().compression_method) {
-            (_, CompressionMethod::ModernJPEG) => self.read_jpeg()?,
```

Comment:
Now that read_stripped_image / read_tiled_image work with jpeg images, could you replace the contents with something like this (perhaps with a check that the image actually uses jpeg compression):
```rust
    #[deprecated = "Use read_image instead"]
    pub fn read_jpeg(&mut self) -> TiffResult<DecodingResult> {
        self.read_image()
    }
```